src/control/cmd/daos/pool.go

```diff
@@ -295,21 +295,36 @@ func convertPoolInfo(pinfo *C.daos_pool_info_t) (*daos.PoolInfo, error) {
 	return poolInfo, nil
 }
 
-func queryPoolRankLists(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*daos.PoolInfo, error) {
-	var rlPtr **C.d_rank_list_t = nil
-	var rl *C.d_rank_list_t = nil
-
-	if queryMask.HasOption(daos.PoolQueryOptionEnabledEngines) || queryMask.HasOption(daos.PoolQueryOptionDisabledEngines) ||
-		queryMask.HasOption(daos.PoolQueryOptionDeadEngines) {
-		rlPtr = &rl
-	}
+func queryPool(poolHdl C.daos_handle_t, queryMask daos.PoolQueryMask) (*daos.PoolInfo, error) {
+	var enabledRanks *C.d_rank_list_t
+	var disabledRanks *C.d_rank_list_t
+	defer func() {
+		C.d_rank_list_free(enabledRanks)
+		C.d_rank_list_free(disabledRanks)
+	}()
 
+	var rc C.int
 	cPoolInfo := C.daos_pool_info_t{
 		pi_bits: C.uint64_t(queryMask),
 	}
+	if queryMask.HasOption(daos.PoolQueryOptionEnabledEngines) && queryMask.HasOption(daos.PoolQueryOptionDisabledEngines) {
+		enaQm := queryMask
+		enaQm.ClearOptions(daos.PoolQueryOptionDisabledEngines)
```

Comment:
how does this work? `queryMask` type `daos.PoolQueryMask` is alias of  `C.uint64_t`. is that somehow a reference? will clearing options here somehow update `cPoolInfo.pi_bits`, I just don't see how? thanks in advance

Reply:
D'oh, good catch. I rushed this last night after abandoning the other approach. You're right; this won't do what I want it to do. I need to update the `pi_bits` field with the new mask.

Reply:
The funny thing is, it worked OK as it is in testing, but that's probably by accident. Oh, well. Definitely best to update it so that we're not relying on implicit behavior.